submission_runner.py

```diff
@@ -300,6 +301,8 @@ def train_once(
     except spec.TrainingCompleteError:
       train_state['training_complete'] = True
     global_step += 1
+    if global_step == max_global_steps:
```

Comment:
do we also need a check for `if max_global_steps is not None and global_step == max_global_steps:` ?

Reply:
If max_global_steps is None then global_step == max_global_steps will be False. So I don't think we need to add an extra check.